website/docs/language/state/encryption.mdx

```diff
@@ -0,0 +1,115 @@
+---
+description: >-
+  Encrypt your state-related data at rest.
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+import ConfigurationTF from '!!raw-loader!./examples/encryption/configuration.tf'
+import ConfigurationSH from '!!raw-loader!./examples/encryption/configuration.sh'
+import ConfigurationPS1 from '!!raw-loader!./examples/encryption/configuration.ps1'
+import Enforce from '!!raw-loader!./examples/encryption/enforce.tf'
+import AESGCM from '!!raw-loader!./examples/encryption/aes_gcm.tf'
+import Fallback from '!!raw-loader!./examples/encryption/fallback.tf'
+import RemoteState from '!!raw-loader!./examples/encryption/terraform_remote_state.tf'
+
+# State and Plan Encryption
+
+OpenTofu supports encrypting state and plan files at rest, both for local storage and when using a backend. In addition, you can also use encryption with the `terraform_remote_state` data source. This page explains how to set up encryption and what encryption method is suitable for which use case.
+
+## General guidance and pitfalls (please read)
+
+When you enable encryption, your state and plan files become unrecoverable without the appropriate encryption key. Please make sure you read this section carefully before enabling encryption.
+
+### What does encryption protect against?
+
+When you enable encryption, OpenTofu will encrypt state data *at rest*. If an attacker were to gain access to your state file, they should not be able to read it and use the sensitive values (e.g. access keys) contained in the state file.
+
+However, encryption does not protect against data loss (your state file getting damaged) and it also does not protect against replay attack (an attacker using an older state or plan file and tricking you into running it). Additionally, OpenTofu does not and cannot protect the sensitive values in the state file from the person running the `tofu` command.
+
+### What precautions do I need to take?
+
+When you enable encryption, consider who needs access to your state file directly. If you have more than a very small number of people with access needs, you may want to consider running your production `plan` and `apply` runs from a continuous integration system to protect both the encryption key and the sensitive values in your state.
+
+You will also need to decide what kind of key you would like to use. You can either opt for a static passphrase which you manually rotate regularly, or you can choose a key management system. If feasible, we recommend using a key management system that can rotate keys automatically to give you the maximum amount of protection.

```

Comment:
I don't think there's any proper reason to regularly rotate passphrases, is there?

With key derivation we'll get a new encryption key every time anyway.

Reply:
See https://github.com/opentofu/opentofu/pull/1291#discussion_r1512927813 -- there's the same reason to rotate passwords as for phrases, unless additional data is mixed in to the password to make it unique.

Likely there isn't, otherwise you could use a KBKDF in place of a PBKDF and get the same properties, IMHO.